src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/helper/builders/CourseActivityBuilder.kt

```diff
@@ -54,19 +58,57 @@ class CourseActivityBuilder(
   var payRates: List<CourseActivityPayRateBuilder>,
 ) {
   fun build(): CourseActivity =
+    repository?.let {
+      CourseActivity(
+        code = code,
+        program = repository.lookupProgramService(programId),
```

Comment:
If the repository has been injected this must be an integration test so load related entities.